### ISSUE TITLE
phpdocumentor2 problems

### ISSUE THREAD

**Issue by lsces** [ADOdb/ADOdb#138]
One of the nice things about ADOdb has always been the phpdoc annotation which works nicely with phpdocumentor1 and many ide's. Just run a build using the latest phpdocumentor2 and it does not look goo :(
http://phpsurgery.org/ADOdb-docs/index.html is what I have but I think that it is phpdocumentor which is wrong rather than the code base, so has anybody had a look at this recently? All of my code base is as bad so I think I need to drop back to an older phpdocumentor and perhaps customise that rather than looking to re-write all the ADOdb code.

**Comment by mnewnham** [comment]
One of the features of version 5.21 when it is released is that it will be completely re-tagged to comply  with current phpdocumentor standards. All files, classes and functions will have a standards-compliant docblock template inserted. For that reason, we won't be looking at any documentation issues until after that.

Mark

**Comment by lsces** [comment]
I always used the original ADOdb docblock standard as my reference, so I would be interested in seeing just what has to be done to make what is needed to change what is perfectly functional into this alternate docblock standard. I don't see why there should be over 5000 errors on something which is already an establisged sandard and works perfectly with our IDE's!

**Comment by mnewnham** [comment]
As the project has migrated from an 'owner supported' project to a 'community supported' project, we need to implement standards and methodologies that can be understood and used by anyone who wants to contribute.

Whilst Jon's documentor standards were fine for his own use, it was done intermittently at best (I personally contributed a large amount of code and I never wrote a line of documentation). As phpdocumentor1 has been superseded by version 2, it was inevitable that at some point we would have to move versions, and now is as good as any time, with a stable code base underlying the change in documentation.  

My hope is that when people see that the project uses recognizable, up-to-date standards, they will be more inclined to both use and contribute towards it. I would certainly hope that the documentation would be much more usable inside an IDE

**Comment by lsces** [comment]
It would seem that phpdocumentor2 needs to fix a problem before we start on anything https://github.com/phpDocumentor/phpDocumentor2/issues/1339 is a large number of the current error list on this, smarty and the rest of my code base.

**Comment by mnewnham** [comment]
I have not seen this issue in the test conversions of the ADOdb programs I have run

**Comment by lsces** [comment]
On 13/07/15 21:19, Mark Newnham wrote:
> I have not seen this issue in the test conversions of the ADOdb programs
> I have run

So what am I doing wrong with http://phpsurgery.org/ADOdb-docs/ ...
Latest download of ADOdb with the latest phpdocumentor2

http://phpsurgery.org/smarty-docs/ has the same problem.

-- 
Lester Caine - G8HFL
-----------------------------
Contact - http://lsces.co.uk/wiki/?page=contact
L.S.Caine Electronic Services - http://lsces.co.uk
EnquirySolve - http://enquirysolve.com/
Model Engineers Digital Workshop - http://medw.co.uk
Rainbow Digital Media - http://rainbowdigitalmedia.co.uk

**Comment by mnewnham** [comment]
The current release of ADOdb is 5.19.  The  phpdocumentor2  compatible release (5.21) is not yet available. This will have every file-level docblock rewritten and every missing class and function level docblock inserted.

**Comment by lsces** [comment]
On 13/07/15 22:05, Mark Newnham wrote:
> The current release of ADOdb is 5.19. The phpdocumentor2 compatible
> release (5.21) is not yet available. This will have every file-level
> docblock rewritten and every missing class and function level docblock
> inserted.

Mark ... what I am asking is just what has to be done to achieve that.
*I* have no problem running an automated script to carry out the bulk of
the work. I've done that before, and I need to do that now on my own
code, but the fixes that I've been given so far do not work because of
the problem with phpdocumentor2 ... If someone can tell me just what
WILL work I'll get on with the job ...

-- 
Lester Caine - G8HFL
-----------------------------
Contact - http://lsces.co.uk/wiki/?page=contact
L.S.Caine Electronic Services - http://lsces.co.uk
EnquirySolve - http://enquirysolve.com/
Model Engineers Digital Workshop - http://medw.co.uk
Rainbow Digital Media - http://rainbowdigitalmedia.co.uk

**Comment by mnewnham** [comment]
We seem to be in a confused state here.  What I am stating is: When 5.21 is released, It will be phpdocumentor2 compliant. Any modifications to the ADOdb code in order for this to have happened will have been done by me. I am fixing it, testing it and running it against phpdocumentor2 to ensure its compliance. What will be missing are substantial numbers of things like parameter lists and  return parameters because those items are missing from the existing documentation,  may not be automatically discernible and will require manual intervention in order to fix, but the files will include all the necessary tags to ensure compliance. Fixing that data will take a while and I do not want to hold up the release schedule in order to complete that work, which will likely take months.

Until that release. I can't offer any guarantee that the code will be compliant with phpdocumentor1 or phpdocumentor2

**Comment by lsces** [comment]
On 13/07/15 22:59, Mark Newnham wrote:
> Until that release. I can't offer any guarantee that the code will be
> compliant with phpdocumentor1 or phpdocumentor2

The code is and always has been PD1 'compliant' and works happily with
IDE's. We have always maintained copies of ADOdb across various
projects, and fed back fixes to the master copy where necessary. YES a
lot of detail is missing on the PD1 documentation, but on the whole PD1
filled in enough detail for a working set of docs. Many of the 'errors'
flagged by PD2 can be ignored although adding extra descriptions later
may be appropriate. Where that has been useful up until now it has
happened, but I think some ports of ADOdb probably have not back ported
those additions :(

PD2 seems to be the problem and having applied the patch on bug #1339
then I've got a cleaner set of docs on
http://phpsurgery.org/ADOdb-docs/index.html. I need to produce a clone
of the original templates from PD1 so I can get back the full hierarchy
and source code cross reference that PD1 supplies, but this is a better
base to build on. Now that I have a more practical starting point I'll
get back to http://lsces.org.uk/wiki/bitweaver+documentation and sort
out if there is a better template to use as a base for the next stage.

I do need to sync http://hg.lsces.org.uk/bw/bitweaver/externals/adodb/
with github. Just another job on the todo list ...

-- 
Lester Caine - G8HFL
-----------------------------
Contact - http://lsces.co.uk/wiki/?page=contact
L.S.Caine Electronic Services - http://lsces.co.uk
EnquirySolve - http://enquirysolve.com/
Model Engineers Digital Workshop - http://medw.co.uk
Rainbow Digital Media - http://rainbowdigitalmedia.co.uk

**Comment by dregad** [comment]
I just tried to run the latest version of phpDocumentor (3.1.1) and whilst the output is not all that great because there is still a lot of work to be done to improve our PHPDoc blocks, API documentation was generated successfully.

Considering that phpDocumentor 3 was released late 2020, I don't see much point in trying to address v2 issues from 6 years ago which I'm sure got fixed or you managed to work around since then :wink: 

A phpDocumentor config file will be included in the next release.
